api/openapi-spec.yaml

```
@@ -0,0 +1,352 @@
+      operationId: getEventsUsingGET
+      parameters:
+      - name: pageSize
+        in: query
+        description: "The number of events to be displayed per page."
+        schema:
+          type: integer
+          format: int32
+          default: 500
+      - name: params
+        in: query
+        description: |
+          To search for specific events or artifacts, filtering with parameters is supported.
+
+          **Syntax:**
+
+          `<resource>?key[.key ...]=value[&key[.key ...]=value ...]`
+
+          To traverse into nested structures and filter on their keys, namespacing with
+          `.` (dot) is used.
+
+          **Examples, single key:**
+
+          `/events?meta.type=EiffelActivityStartedEvent`
+
+          **Examples multiple keys:**
+
+          ```
+          /events?key1=value1&key2=value2
+          /events?key1=value1&key2=value2&key3=value3
+          ```
+
+          **Examples nested structures:**
+          ```
+          /events?data.identity=pkg:maven/my.namespace/my-name@1.0.0
+          /events?meta.source.domainId=my.domain&data.identity=pkg:maven/my.namespace/my-name@1.0.0    #Multiple keys and nested structures
+          ```
+
+          Note that mutiple keys only is allowed with logical AND (via `&`). There are no support for logical OR.
+
+          **Data types:**
+
+          By default, all data types are treated as strings. It is possible to specify the data type by including it explicitly in the query:
+
+          ```
+          /events?int(key1)>=value1&key2=value2&double(key3)=value3
+          ```
+
+          In this example, `key1` and `value1` are treated as integers, `key2` and `value2` are treated as strings and `key3` and `values3` are treated as doubles.
+
+          Currently only string (default), integer, long and double data types are supported.
+
+          **No comparator:**
+
+          Note that for events query, there is one special supported case is that the query expression has no comparator and value at all, for example `/events?key`, which means getting all events who has `key` as a field in the JSON documents. The REST API for events query also supports `!key` as a filter which means getting all events without field ‘key’ in the JSON documents. Here are some examples:
+
+          ```
+          /events?data.identity                                   #Get all the events containing field 'identity' in the 'data'
+          /events?meta.source.domainId=my.domain&data.identity    #DomainId is 'my.domain' and has field 'identity' in 'data'
+          /events?!data.customData                                #Return all the events without field 'customData' in 'data'
+          ```
+
+          **Example**
+
+          The following example adds two search parameters. The result of the object will result in adding `meta.type=EiffelArtifactCreatedEvent&data.identity=pkg%3Amaven%2Fmy.namespace%2Fmy-name%401.0.0` to the search.
```

This example looks weird, or the text around it is strangely written. What is the example and what result is it that will result in something?

```suggestion
          The example below shows how to write a query using multiple keys, similar to what has been described earlier, but using a request body rather than a url query string. The request body below will result the following query string:  `meta.type=EiffelArtifactCreatedEvent&data.identity=pkg%3Amaven%2Fmy.namespace%2Fmy-name%401.0.0`
```

---

Tried to add some references (in [7087272](https://github.com/eiffel-community/eiffel-event-repository/pull/12/commits/7087272846ee81cb3160abd56bf6747da5bc8af7)) as this seems to be the way to give an example of multiple query parameters in OpenAPI

---

Improved in

---

My confusion was rather on that the actual example is outside the description property. I wrote a separate comment about that, so this comment can be neglected now.